flutter/test/mocks.dart

```diff
@@ -206,3 +206,39 @@ final fakeFrameDurations = [
   Duration(milliseconds: 40),
   Duration(milliseconds: 710),
 ];
+
+@GenerateMocks([Callbacks])
+abstract class Callbacks {
+  Future<Object?>? methodCallHandler(String method, [dynamic arguments]);
+}
+
+class NativeChannelFixture {
+  late final MethodChannel channel;
+  late final Future<Object?>? Function(String method, [dynamic arguments])
+      handler;
+  static TestDefaultBinaryMessenger get _messenger =>
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger;
+
+  NativeChannelFixture() {
+    TestWidgetsFlutterBinding.ensureInitialized();
+    channel = MethodChannel('test.channel', StandardMethodCodec(), _messenger);
+    // channel = MockMethodChannel();
+    // when(channel.name).thenReturn('test.channel');
+    // when(channel.binaryMessenger).thenReturn(_messenger);
+    handler = MockCallbacks().methodCallHandler;
+    // _messenger.setMessageHandler(channel.name, (ByteData? data) async {
+    //   final call = StandardMethodCodec().decodeMethodCall(data);
+    //   await handler(call);
+    // });
```

Comment:
do we need those commented lines later on?

Reply:
my bad, missed this in self-review